lib/datadog/core/utils/url.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Core
+    module Utils
+      # Helpers class that provides methods to process URLs
+      # such as filtering sensitive information.
+      module Url
+        def self.filter_sensitive_info(url)
```

Comment:
@marcotc or @TonyCTHsu do we not already have something to handle scrubbing sensitive info from urls?

Reply:
If we are only concerned with basic auth, I would prefer leveraging `URI` object instead regex.

```ruby
          URI.parse(url).tap do |u|
            u.user = nil
            u.password = nil
          end.to_s
```

This implement is only failing the test example `without protocol and without credentials`, but I think raising `URI::InvalidURIError` is correct since it is indeed not a URI.

Reply:
BTW, it looks like `Datadog::Tracing::Contrib::Utils::Quantization` does not contain functionality to filter out basic auth info.

Reply:
yeah, we can use URI approach here, as long as I make the specification tests pass